src/modules/form/components/DynamicFormField.tsx

```diff
@@ -0,0 +1,272 @@
+              if (!item) return null;
+              return (
+                <AutofillFormItemWrapper handlers={handlers} item={item}>
+                  {(autofillValue) => {
+                    return isCurrency
+                      ? formatCurrency(autofillValue || 0)
+                      : autofillValue || 0;
+                  }}
+                </AutofillFormItemWrapper>
+              );
+            }}
+            values={values}
+            itemChanged={itemChanged}
+            severalItemsChanged={severalItemsChanged}
+            visible={visible}
+            debug={
+              import.meta.env.MODE === 'development'
+                ? (keys?: string[]) => {
+                    const sectionValues = keys ? pick(values, keys) : values;
+                    const valuesByKey = transformSubmitValues({
+                      definition,
+                      values: sectionValues,
+                      keyByFieldName: true,
+                    });
+                    // eslint-disable-next-line no-console
+                    console.group(item.text || item.linkId);
+                    // eslint-disable-next-line no-console
+                    console.log(sectionValues);
+                    // eslint-disable-next-line no-console
+                    console.log(valuesByKey);
+                    // eslint-disable-next-line no-console
+                    console.groupEnd();
+                  }
+                : undefined
+            }
+          />
+        );
+
+        // Disability group actually needs accurate values for its own mechanics, so provide them
+        if (item.component === Component.DisabilityTable) {
+          return (
+            <ValueWrapper
+              handlers={handlers}
+              name={getAllChildLinkIds(item).map(getSafeLinkId)}
+            >
+              {/* We're just using this component to watch the group's child values and update the values prop when they change */}
+              {() =>
+                cloneElement(group, { values: handlers.getCleanedValues() })
+              }
+            </ValueWrapper>
+          );
+        }
+
+        return group;
+      }
+
+      const itemComponent = item.readOnly ? (
+        <DynamicViewField
+          item={item}
+          value={values[item.linkId]}
+          disabled={isDisabled}
+          nestingLevel={nestingLevel}
+          horizontal={horizontal}
+          pickListArgs={pickListArgs}
+          // Needed because there are some enable/disabled and autofill dependencies that depend on PickListOption.labels that are fetched (PriorLivingSituation is an example)
+          adjustValue={itemChanged}
+          {...fieldProps}
+        />
+      ) : (
+        <ValueWrapper name={getSafeLinkId(item.linkId)} handlers={handlers}>
+          {(value) => (
+            <DynamicField
+              value={value}
+              key={item.linkId}
+              item={item}
+              itemChanged={itemChanged}
```

Comment:
Note to self:

ValueWrapper provides value for this form item from RHF state; `itemChanged` uses `setValue` internally to update the form state. This lets us use all the existing input components and `DynamicField` without modification.

(As opposed to adjusting DynamicField, or the input components themselves, to consume the RHF context)